src/handler_forward.js

```diff
@@ -16,9 +17,104 @@ export default class HandlerForward extends HandlerBase {
     }
 
     run() {
+        /*
+        * Here we decide if we should create a tls socket or just forward the request
+        * */
+        if (this.upstreamProxyUrlParsed.scheme === 'http') {
+            this.forwardRequest({});
+            return;
+        }
+
+        const tlsSocket = tls.connect({
+            host: this.upstreamProxyUrlParsed.hostname,
+            port: this.upstreamProxyUrlParsed.port
+        });
+
+        const headers = { ...this.proxyHeaders };
+        maybeAddProxyAuthorizationHeader(this.upstreamProxyUrlParsed, headers);
+        const requestUrl = new URL(this.srcRequest.url);
+        const hostname = `${requestUrl.hostname}:${requestUrl.port || '80'}`;
+
+        let payload = `CONNECT ${hostname} HTTP/1.1\r\n`;
```

Comment:
Using the CONNECT protocol is unnecessary. This is just a regular web proxy.

https://github.com/apify/proxy-chain/blob/0043ac9db2c04fffdd53fe7ee7d3a11183648555/src/handler_forward.js#L114

```js
const fn = reqOpts.protocol === 'https:' ? https.request : http.request;
this.trgRequest = fn(reqOpts); 
```

and it should work.

Reply:
Ah, I just realized the upstream proxy server doesn't have to support HTTPS...